File: examples/redis/topologies/wordcount-mem.clj
(ns wordcount-mem
  (:use     [streamparse.specs])
  (:gen-class))

(defn wordcount-mem [options]
   [
    ;; spout configuration
    {"word-spout" (python-spout-spec
          options
          "wordcount_mem.WordSpout"
          ["word"]
          )
    }
    ;; bolt configuration
    {"count-bolt" (python-bolt-spec
          options
          ;; field group on word due to memory storage
          {"word-spout" ["word"]}
          "wordcount_mem.WordCountBolt"
          ["word" "count"]
          :p 2
          )
    }
  ]
)

File: examples/redis/topologies/wordcount_mem.py
"""
Word count topology (in memory)
"""

from bolts import WordCountBolt
from spouts import WordSpout
from streamparse import Grouping, Topology


class WordCount(Topology):
    word_spout = WordSpout.spec()
    count_bolt = WordCountBolt.spec(inputs={word_spout: Grouping.fields("word")}, par=2)
